core/src/impl/Kokkos_Profiling.hpp

```diff
@@ -303,21 +333,68 @@ void tune_policy(const size_t /**tuning_context*/, const std::string& label_in,
       label = name.get();
     }
     auto tuner_iter = [&]() {
-      auto my_tuner = team_tuners.find(label);
-      if (my_tuner == team_tuners.end()) {
-        return (
-            team_tuners
-                .emplace(label,
-                         Kokkos::Tools::Experimental::TeamSizeTuner(
-                             label, policy, functor, tag,
+      auto my_tuner = map.find(label);
+      if (my_tuner == map.end()) {
+        return (map.emplace(
+                       label,
+                       Tuner(label, policy, functor, tag,
                              Impl::ComplexReducerSizeCalculator<ReducerType>{}))
-                .first);
+                    .first);
       }
       return my_tuner;
     }();
     tuner_iter->second.tune(policy);
   }
 }
+template <class Functor, class TagType, class... Properties>
+void tune_policy(const size_t /**tuning_context*/, const std::string& label_in,
+                 Kokkos::TeamPolicy<Properties...>& policy,
+                 const Functor& functor, const TagType& tag) {
+  generic_tune_policy<Experimental::TeamSizeTuner>(
+      label_in, team_tuners, policy, functor, tag,
+      [](const Kokkos::TeamPolicy<Properties...>& candidate_policy) {
+        return (candidate_policy.impl_auto_team_size() ||
+                candidate_policy.impl_auto_vector_length());
+      });
+}
+
+template <class ReducerType, class Functor, class TagType, class... Properties>
+void tune_policy(const size_t /**tuning_context*/, const std::string& label_in,
+                 Kokkos::TeamPolicy<Properties...>& policy,
+                 const Functor& functor, const TagType& tag) {
```

Comment:
Document these overloads (single line above each of them would do).  I had to stare at their signature wondering what was different.

Reply:
065fead